core/src/main/scala/codesearch/core/index/CratesSources.scala

```diff
@@ -1,58 +1,41 @@
 package codesearch.core.index
 
-import ammonite.ops.{Path, pwd}
+import codesearch.core.index.repository.CratesPackage
 import codesearch.core.model.CratesTable
-import codesearch.core.util.Helper
 import org.slf4j.{Logger, LoggerFactory}
 
-import scala.collection.mutable
-import scala.sys.process._
 import scala.concurrent.Future
 
 object CratesSources extends Sources[CratesTable] {
-  val SOURCES: Path = pwd / 'data / 'rust / 'packages
 
-  private val CARGO_PATH = "./cargo"
-
-  override val logger: Logger = LoggerFactory.getLogger(CratesSources.getClass)
+  override val logger: Logger             = LoggerFactory.getLogger(CratesSources.getClass)
   override val indexAPI: CratesIndex.type = CratesIndex
-  override val indexFile: String = ".crates_csearch_index"
-  override val langExts: String = ".*\\.(rs)$"
-
-
-  def csearch(searchQuery: String, insensitive: Boolean, precise: Boolean, sources: Boolean, page: Int = 0): Future[(Int, Seq[PackageResult])] = {
+  override val indexFile: String          = ".crates_csearch_index"
+  override val langExts: String           = ".*\\.(rs)$"
+
+  def csearch(searchQuery: String,
+              insensitive: Boolean,
+              precise: Boolean,
+              sources: Boolean,
+              page: Int = 0): Future[(Int, Seq[PackageResult])] = {
     val answers = runCsearch(searchQuery, insensitive, precise, sources)
     indexAPI.verNames().map { verSeq =>
       val nameToVersion = Map(verSeq: _*)
-      (answers.length, answers
-        .slice(math.max(page - 1, 0) * 100, page * 100)
-        .flatMap(uri => indexAPI contentByURI(uri, nameToVersion)).groupBy(x => (x._1, x._2)).map {
-        case ((name, packageLink), results) =>
-          PackageResult(name, packageLink, results.map(_._3))
-      }.toSeq.sortBy(_.name))
+      (answers.length,
+       answers
+         .slice(math.max(page - 1, 0) * 100, page * 100)
```

Comment:
May be better declare PAGE_SIZE in `Sources` scope or whatever in order to avoid logical or mistype errors in the future.